app/[locale]/og/route.tsx

```diff
@@ -0,0 +1,72 @@
+import { ImageResponse } from 'next/og';
+
+import JsIconWhite from '@/components/Icons/Logos/JsIconWhite';
+import tailwindConfig from '@/tailwind.config';
+
+export const runtime = process.env.NEXT_PUBLIC_VERCEL_URL ? 'edge' : false;
+
+function hexToRgba(hex: string, alpha = 0.9) {
+  hex = hex.replace(/^#/, '');
+
+  const bigint = parseInt(hex, 16);
+  const r = (bigint >> 16) & 255;
+  const g = (bigint >> 8) & 255;
+  const b = bigint & 255;
+
+  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
+}
+
+export async function GET(request: Request) {
+  try {
+    const { searchParams, host, protocol } = new URL(request.url);
+
+    // ?title=<title>
+    const hasTitle = searchParams.has('title');
+    const title = hasTitle
+      ? searchParams.get('title')?.slice(0, 100)
+      : undefined;
+
+    //?type=<type>
+    const type = searchParams.get('type') ?? 'announcement';
+
+    const typeAttributes: { [key: string]: string } = {
+      announcement: tailwindConfig.theme.colors.green['700'],
+      release: tailwindConfig.theme.colors.info['600'],
+      vulnerability: tailwindConfig.theme.colors.warning['600'],
+    };
+
+    return new ImageResponse(
+      (
+        <div
+          tw={`relative flex items-center justify-center bg-black
```

Comment:
note - this OG tailwind technique cannot use our styles, it must use the static `tw` attribute.
That means I couldn't simply re-use `Preview`